tests/unit/amo/components/TestCollectionAddAddon.js

```diff
@@ -0,0 +1,262 @@
+    clock = sinon.useFakeTimers(Date.now());
+  });
+
+  afterEach(() => {
+    clock.restore();
+  });
+
+  const dispatchSignedInUser = ({
+    userId = signedInUserId,
+    username = signedInUsername,
+  }) => {
+    return dispatchSignInActions({
+      userId,
+      userProps: { username },
+    });
+  };
+
+  const getProps = ({
+    collection = createInternalCollection({
+      detail: createFakeCollectionDetail(),
+    }),
+    ...customProps
+  }) => {
+    return {
+      collection,
+      filters: {},
+      i18n: fakeI18n(),
+      store: dispatchClientMetadata().store,
+      ...customProps,
+    };
+  };
+
+  const render = (customProps = {}) => {
+    const props = getProps(customProps);
+    return shallowUntilTarget(
+      <CollectionAddAddon {...props} />,
+      CollectionAddAddonBase,
+    );
+  };
+
+  it('renders any error', () => {
+    const errorHandler = createStubErrorHandler(
+      new Error('Unexpected API error'),
+    );
+    const root = render({ errorHandler });
+
+    expect(root.find(ErrorList)).toHaveLength(1);
+  });
+
+  it('dispatches addAddonToCollection when selecting an add-on', () => {
+    const { store } = dispatchSignedInUser({
+      username: signedInUsername,
+    });
+    const filters = { page: 2 };
+    const errorHandler = createStubErrorHandler();
+
+    const collection = createInternalCollection({
+      detail: createFakeCollectionDetail({ authorUsername: signedInUsername }),
+    });
+    const dispatchSpy = sinon.spy(store, 'dispatch');
+    const root = render({ collection, errorHandler, filters, store });
+
+    const suggestion = createInternalSuggestion(
+      createFakeAutocompleteResult({ name: 'uBlock Origin' }),
+    );
+    const selectSuggestion = simulateAutoSearchCallback({
+      root,
+      propName: 'onSuggestionSelected',
+    });
+    selectSuggestion(suggestion);
+
+    sinon.assert.calledWith(
+      dispatchSpy,
+      addAddonToCollection({
+        addonId: suggestion.addonId,
+        collectionId: collection.id,
+        slug: collection.slug,
+        editing: true,
+        errorHandlerId: errorHandler.id,
+        filters,
+        username: signedInUsername,
+      }),
+    );
+  });
+
+  it('sets the addonWasAdded state to false when selecting an add-on', () => {
+    const { store } = dispatchSignedInUser({
+      username: signedInUsername,
+    });
+    const root = render({ store });
+
+    setAddonWasAddedUiStateToTrue({ username: signedInUsername, root, store });
+    expect(root.instance().props.uiState.addonWasAdded).toEqual(true);
+
+    const suggestion = createInternalSuggestion(
+      createFakeAutocompleteResult({ name: 'uBlock Origin' }),
+    );
+    const selectSuggestion = simulateAutoSearchCallback({
+      root,
+      propName: 'onSuggestionSelected',
+    });
+    selectSuggestion(suggestion);
+
+    applyUIStateChanges({ root, store });
+    expect(root.instance().props.uiState.addonWasAdded).toEqual(false);
```

Comment:
This is testing the implementation of the component rather than the functionality of the component. I think it's better to test functionality because it lets the implementation evolve freely. 

I think you can actually just delete this test. The functionality is already tested in *it removes the notification after a new add-on has been selected*.